components/jira/actions/create-issue/create-issue.mjs

```diff
@@ -1,228 +1,213 @@
-// legacy_hash_id: a_0Mi2KV
-import { axios } from "@pipedream/platform";
+import jira from "../../jira.app.mjs";
 
 export default {
   key: "jira-create-issue",
   name: "Create Issue",
-  description: "Creates an issue or, where the option to create subtasks is enabled in Jira, a subtask.",
-  version: "0.1.2",
+  description: "Creates an issue or, where the option to create subtasks is enabled in Jira, a subtask, [See the docs](https://developer.atlassian.com/cloud/jira/platform/rest/v3/#api-rest-api-3-issue-post)",
+  version: "0.1.3",
   type: "action",
   props: {
-    jira: {
-      type: "app",
-      app: "jira",
+    jira,
+    projectID: {
+      propDefinition: [
+        jira,
+        "projectID",
+      ],
+    },
+    issueType: {
+      propDefinition: [
+        jira,
+        "issueType",
+        (configuredProps) => ({
+          projectID: configuredProps.projectID,
+        }),
+      ],
     },
     summary: {
       type: "string",
-      label: "Issue Summary",
+      label: "Issue summary",
       description: "The title of the issue",
     },
-    parent_key: {
-      type: "string",
-      description: "Key of the project where the issue to create pertains.",
-      optional: true,
+    description: {
+      type: "object",
+      label: "Description",
+      description: "Description object of the issue, Jira accepts `doc` type of descriptions, [See Atlassian Document Structure](https://developer.atlassian.com/cloud/jira/platform/apis/document/structure/)",
     },
-    issuetype: {
+    parentKey: {
       type: "string",
-      label: "Issue Type",
-      description: "An ID identifying the type of issue you'd like to create. See the API docs at https://developer.atlassian.com/cloud/jira/platform/rest/v3/#api-rest-api-3-issue-post to see available options",
+      label: "Parent key",
+      description: "Key of the project where the issue to create pertains.",
       optional: true,
     },
     components: {
       type: "any",
+      label: "Components",
       description: "Array of components of the issue.",
       optional: true,
     },
-    projectID: {
-      type: "string",
-      description: "The ID of the project where the issue will be created.",
-    },
-    description: {
-      type: "object",
-      description: "Description object of the issue.",
-    },
-    reporter_id: {
+    reporterId: {
       type: "string",
-    },
-    fixVersions: {
-      type: "any",
-      optional: true,
+      label: "Reporter ID",
+      description: "Reporter ID of the issue.",
     },
     labels: {
       type: "any",
+      label: "Labels",
       description: "Labels associated to the issue, such as \"bugfix\", \"blitz_test\".",
       optional: true,
     },
```

Comment:
We should use async options to get the available `labels` for the user to select from, and move this prop to the app file.
https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-labels/#api-rest-api-3-label-get

Reply:
Actually I did not realize it, good catch thanks!